Sprint-1/destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +70,33 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+// Task 1:
+// Function to display names of people in Gryffindor house
+function displayGryffindorMembers(hogwarts) {
```

Comment:
You are creating this function and parameters. What do you think displayGryffindorMembers() will accept as parameters? Sure, when you will invoke this method, you will pass hogwarts to it (as this variable is defined for you), but it is up to you to come up with a good name for the parameters of your function. Think what this parameter will hold as a value and name it accordingly. This is not a show-stopper for your PR but it is good to practice.
Next question, what will happen if I will do something like this:

displayGryffindorMembers(null);

Reply:
Hi @tyzia,

Thank you for your feedback! I hadn’t considered it before, but I’ve made the improvements. I added a check at the start of each function to ensure the input is valid. If the input is null, undefined, or not an array, the function will show an error and stop. I also renamed the parameter from hogwarts to hogwartsMembers, making it clearer and more descriptive. 😊

When I called displayGryffindorMembers(null);, I got a TypeError. Inside the function, the code attempts to execute hogwarts.forEach(...), but .forEach() can only be called on an array. Since null is not an array, this throws a TypeError.

Thanks again. I’ve learned a lot! 🌸